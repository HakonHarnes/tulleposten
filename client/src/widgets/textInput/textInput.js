// @flow
import * as React from 'react';
import { Component } from 'react-simplified';

import style from './css/text-input.module.css';

class TextInputLight extends Component<{
   onChange: any => mixed,
   className: string,
   placeholder?: string,
   value?: string,
   required: boolean
}> {
   render() {
      return (
         <input
            data-testid="text-input-light"
            className={`${this.props.className} ${style['text-input']} ${style['text-input-light']}`}
            type="text"
            placeholder={this.props.placeholder}
            value={this.props.value}
            onChange={this.props.onChange}
            required={this.props.required}
         />
      );
   }
}

class TextInputDark extends Component<{
   onChange: any => mixed,
   className: string,
   placeholder?: string,
   value?: string,
   required: boolean
}> {
   render() {
      return (
         <input
            data-testid="text-input-dark"
            className={`${this.props.className} ${style['text-input']} ${style['text-input-dark']}`}
            type="text"
            placeholder={this.props.placeholder}
            value={this.props.value}
            onChange={this.props.onChange}
            required={this.props.required}
         />
      );
   }
}

export class TextInput {
   static Light = TextInputLight;
   static Dark = TextInputDark;
}
