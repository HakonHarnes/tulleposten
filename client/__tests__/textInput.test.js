//@flow
import * as React from 'react';
import ReactDOM from 'react-dom';

import '@testing-library/jest-dom/extend-expect';

import { render, fireEvent, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { TextInput } from '../src/widgets/textInput/textInput';

const changeHandler = jest.fn();

afterEach(cleanup);

it('renders without crashing', () => {
   const div = document.createElement('div');
   ReactDOM.render(<TextInput.Light onChange={changeHandler} className="input" required={false} />, div);
   ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
   const { getByTestId } = render(<TextInput.Light onChange={changeHandler} value="Vetle" className="input" required={true} />);
   expect(getByTestId('text-input-light')).toHaveValue('Vetle');
   expect(getByTestId('text-input-light')).toBeRequired();
   expect(getByTestId('text-input-light')).toHaveClass('input');
});

it('matches snapshot', () => {
   const tree = renderer.create(<TextInput.Light onChange={changeHandler} value="Vetle" className="input" required={false} />).toJSON();
   expect(tree).toMatchSnapshot();
});
