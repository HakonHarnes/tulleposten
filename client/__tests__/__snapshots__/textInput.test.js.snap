// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<input
  className="input text-input text-input-light"
  data-testid="text-input-light"
  onChange={[MockFunction]}
  required={false}
  type="text"
  value="Vetle"
/>
`;
